wsgi/hpack.cpp

```diff
@@ -296,7 +296,7 @@ int HPack::decode(unsigned char *it, unsigned char *itEnd, H2Stream *stream)
         if (*it & 0x80){
             it = decodeUInt16(it, itEnd, intValue, INT_MASK(7));
 //            qDebug() << "6.1 Indexed Header Field Representation" << *it << intValue << it;
-            if (!it || intValue <= 0) {
+            if (!it || intValue == 0) {
```

Comment:
Why is this?

Reply:
cppcheck said n unsigned variable cannot be negative... I'm fine wit postponing to Cutelyst 3

Reply:
ah ok, well we don't need to postpone everything, I think mostly the destructors removal need to be postponed

Reply:
but ABI will change, if we remove the d_ptr, right?

Reply:
ah yes, well if you could split the PR we could apply the safe parts and leave the PR that breaks ABI open till v3

Reply:
Removing the destructor and d_ptr from Action classes is probably fine now, because they are dynamic loaded or used mostly internally

Reply:
The View classes are used by client code and that would bring issues